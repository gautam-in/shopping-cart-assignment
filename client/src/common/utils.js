const PHONE_NUMBER_LENGTH = 10;

export function getUrlParameter(name) {
  name = name.replace(/[[]/, '\\[').replace(/[\]]/, '\\]');
  const regex = new RegExp(`[\\?&]${name}=([^&#]*)`);
  const { location } = window;
  const results = regex.exec(location.search);
  return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
}

export const getExtension = (path) => {
  const fullFileName = path.substring(path.lastIndexOf('/') + 1);
  return fullFileName.substring(fullFileName.lastIndexOf('.') + 1).toLowerCase();
};

export const isEmailValid = (email) => {
  const emailPattern = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,4})+$/;
  return emailPattern.test(email);
};

export const isUsernameValid = (username) => username.length >= USERNAME_MIN_LENGTH;

export const isPhoneValid = (phone) => phone.length === PHONE_NUMBER_LENGTH;
