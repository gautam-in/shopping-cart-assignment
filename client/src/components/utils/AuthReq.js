const API_URL = "http://localhost:8080/auth/";
class AuthReq{
  
  login(email, password) {
    return fetch(API_URL + "login", {
      method: "POST",
      headers:{
        'Content-Type': 'application/x-www-form-urlencoded'
      },
      body: `email=${email}&password=${password}`,
      mode: "cors"
    })
    .then(response => {
      if (response.ok) 
        return response.json()
      return response.json()
    }).then(data =>{
      localStorage.setItem("user", JSON.stringify(data));
      return data;
    })
  }
  
  register(email, password, cpassword) {
    return fetch(API_URL + "register", {
      method: "POST",
      headers:{ 
        'Content-Type': 'application/x-www-form-urlencoded'
      },
      body: `email=${email}&password=${password}&cpassword=${cpassword}`,
      mode: "cors"
    }).then(response => response.json())
      .then(data => data)
      .catch(err => err);
  }

  logout() {
    localStorage.removeItem("user");
  }

  getCurrentUser() {
    return JSON.parse(localStorage.getItem('user'));;
  }
}

export default new AuthReq();