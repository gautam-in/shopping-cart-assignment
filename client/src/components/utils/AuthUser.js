import axios from 'axios';
import authHeader from './auth-header';

const API_URL = 'http://localhost:8080/api/';

class UserService {
  getPublicContent() {
    return axios.get(API_URL + 'all');
  }

  getUserBoard() {
    return axios.get(API_URL + 'user', { headers: authHeader() });
  }
}

export default new UserService();