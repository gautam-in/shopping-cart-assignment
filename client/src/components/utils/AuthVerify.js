import React, { Component } from "react";
import { withRouter } from "react-router-dom";

const parseJwt = (token) => {
  try {
    return JSON.parse(atob(token.split('.')[1]));
  } catch (e) {
    return null;
  }
};

class AuthVerify extends Component {
  constructor(props) {
    super(props);

    props.history.listen(() => {
      const user = JSON.parse(localStorage.getItem("user"));

      if (user) {
        const decodedJwt = parseJwt(user.accessToken);

        if (decodedJwt.exp * 1000 < Date.now()) {
          props.logOut();
        }
      }
    });
  }

  render() {
    return <div></div>;
  }
}

export default withRouter(AuthVerify);