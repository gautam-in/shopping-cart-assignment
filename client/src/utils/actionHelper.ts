interface IBaseActionTypes {
  LOADING: string;
  SUCCESS: string;
  ERROR: string;
}
const actionTypesCreator = (prefix: string, actionType: string): IBaseActionTypes => {
  const types = ["LOADING", "SUCCESS", "ERROR"];
  let data = {
    LOADING: "",
    SUCCESS: "",
    ERROR: ""
  };

  types.forEach((type: string) => {
    let completeActionType = `${prefix}/${actionType}_${type}`;
    Object.assign(data, {
      [type]: completeActionType
    });
  });

  return data;
};
export const ActionHelper = { actionTypesCreator };
