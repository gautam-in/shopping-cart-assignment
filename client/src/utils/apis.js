import axios from 'axios';

const BASE_URL = 'http://localhost:8989/';

export const get = async (url) => {
  try {
    const response = await axios.get(`${BASE_URL}${url}`);
    return response;
  } catch (error) {
    return error;
  }
};
