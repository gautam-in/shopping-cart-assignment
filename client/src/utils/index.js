export const validateFormAndGetErrorMsg = (form, formValues) => {
    let value = ''
    let pattern = ''
    let errors = {}
    form.map((field, i) => {
        if (field.required) {
            value = formValues[field.id]
            pattern = field.pattern
            if(field?.condition) {
                errors = validateCondtionGetErrors(field, formValues, errors)
            }else if (!(pattern.test(value))) {
                errors[field.id] = field.errorMsg
            } else {
                delete errors[field.id]
            }
        }
    })
    return { errors, length: Object.keys(errors).length }
}

export const validateCondtionGetErrors = (field,formValues,errors) => {
    let {key, type} = field.condition
    switch(type) {
        case 'match' : 
            if(formValues[key] !== formValues[field.id]) {
                return {...errors, [field.id]: field.errorMsg}
            }
    }
    return errors
}
export const setInitialValues = (form) => {
    let initialValues = {}
    form.map((item, i) => initialValues[item['id']] = item['value'])
    return initialValues
}

export const updateDocumentTitle = (location) => {
    let { pathname } = location
    let [root, route, id = undefined] = pathname.split("/")
    let brandName = 'Sabka Bazar'
    let title = (root == '' && route == '') ? `${brandName} - Home` : route ? `${brandName} - ${route}` : ''
    document.title = `${title}`;
    if(id) {
        scrollToTop()
    }
    return `/${route}`
}

export const  scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth"
    });
  };

export const FocusToErrorField = (elementsRef, errors) => {
    if(Object.keys(errors).length > 0) {
        elementsRef.current.map((ele) => {
          if(ele.name == Object.keys(errors)[0]) {
            ele.focus()
          }
        })
      }
}
export const isEmpty = (val) => {
    if (val === undefined)
        return true;
    if (typeof (val) == 'function' || typeof (val) == 'number' || typeof (val) == 'boolean' || Object.prototype.toString.call(val) === '[object Date]')
        return false;
    if (val == null || val.length === 0)        // null or 0 length array
        return true;
    if (typeof (val) == "object") {
        // empty object
        var r = true;
        for (var f in val)
            r = false;
        return r;
    }
    return false;
}