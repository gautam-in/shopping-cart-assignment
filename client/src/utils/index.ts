export * from "./TestUtils"

export * from "./firebase"

export * from "./validationSchema"

export const pluralize = (
  word: string,
  suffix: string,
  quantity: number
): string => (quantity === 1 ? `${word}` : `${word}${suffix}`)
