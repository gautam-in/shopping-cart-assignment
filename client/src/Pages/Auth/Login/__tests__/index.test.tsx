import { expect, it } from "vitest"
import { render } from "@testing-library/react"

import { Login } from "../"
import { WrappedComponent } from "../../../../utils"

it("should match Homepage snapshot", () => {
  const wrapper = render(
    <WrappedComponent>
      <Login />
    </WrappedComponent>
  )

  expect(wrapper).toMatchSnapshot()
})
