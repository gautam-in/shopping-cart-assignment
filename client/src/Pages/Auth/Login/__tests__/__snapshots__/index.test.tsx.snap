// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match Homepage snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="login grid"
        id="login"
      >
        <div
          class="description"
        >
          <h1>
            Login
          </h1>
          <p>
            Get access to your orders, Wishlist and Recommendations.
          </p>
        </div>
        <div
          class="login-form"
        >
          <form>
            <div
              class="input-group "
            >
              <input
                aria-invalid="false"
                aria-label="email"
                aria-required="true"
                class="input-control"
                id="email"
                name="email"
                placeholder=" "
                type="email"
                value=""
              />
              <label
                class="input-label"
                for="email"
              >
                Email
              </label>
            </div>
            <div
              class="input-group "
            >
              <input
                aria-invalid="false"
                aria-label="password"
                aria-required="true"
                class="input-control"
                id="password"
                name="password"
                placeholder=" "
                type="password"
                value=""
              />
              <label
                class="input-label"
                for="password"
              >
                Password
              </label>
            </div>
            <button
              class="btn primary "
              disabled=""
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="login grid"
      id="login"
    >
      <div
        class="description"
      >
        <h1>
          Login
        </h1>
        <p>
          Get access to your orders, Wishlist and Recommendations.
        </p>
      </div>
      <div
        class="login-form"
      >
        <form>
          <div
            class="input-group "
          >
            <input
              aria-invalid="false"
              aria-label="email"
              aria-required="true"
              class="input-control"
              id="email"
              name="email"
              placeholder=" "
              type="email"
              value=""
            />
            <label
              class="input-label"
              for="email"
            >
              Email
            </label>
          </div>
          <div
            class="input-group "
          >
            <input
              aria-invalid="false"
              aria-label="password"
              aria-required="true"
              class="input-control"
              id="password"
              name="password"
              placeholder=" "
              type="password"
              value=""
            />
            <label
              class="input-label"
              for="password"
            >
              Password
            </label>
          </div>
          <button
            class="btn primary "
            disabled=""
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
